### PR TITLE
feat(tabs): add select/unselect tabs

### LINKED ISSUE #5: Selecting/Unselecting tabs
**Is your feature request related to a problem? Please describe.**
If we have duplicate tabs or tabs we don't want, we should be able to select a few of them instead of all

**Describe the solution you'd like**
Should be able to select a few tabs or all, or none etc.

**Describe alternatives you've considered**
We considered closing the tab etc, but that's not that scalable I think.

### PR DESCRIPTION
**Pull request by AhsanAyaz**
added the ability to select/unselect tabs
added the type AppTab
removed the ability to go to a page

Fixes #5 

image

### CODE REVIEW

File: src/pages/Popup/Popup.tsx [hunk 1]
```diff
@@ -28,12 +32,48 @@ const Popup = () => {
   }, [repo]);
 
   const sendActionToTabs = (action: any) => {
-    for (let i = 0; i < tabs.length; i++) {
-      const tabId = tabs[i].id as number;
+    tabs.filter(tabItem => (tabItem.isSelected)).forEach(tabItem => {
+      const tabId = tabItem.id as number;
       chrome.tabs.sendMessage(tabId, { action });
-    }
+    })
```
Comment (Skarpej) [inline src/pages/Popup/Popup.tsx:38]: I think you have to move this ')' up one row

File: src/utils.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
+import { AppTab } from "./types/tab.type";
+
 const STORAGE_KEYS = {
   SEARCHED_REPO_NAME: 'gch-searched-repo',
 };
-export const getRepoNameFromUrl = (tab: chrome.tabs.Tab, repo = '') => {
+export const getRepoNameFromUrl = (tab: AppTab, repo = '') => {
   const urlChunks = tab.url!.split('files');
   let repoName = urlChunks[0];
```
Comment (Skarpej) [inline src/utils.ts:8]: Not really related to your commit, but why isn't this a constant variable?

File: src/pages/Popup/Popup.tsx (hunk 1 above)
Comment (AhsanAyaz) [reply src/pages/Popup/Popup.tsx:38]: I've double-checked. This seems appropriate as it is.
```ts
tabs.filter(tabItem => (tabItem.isSelected)).forEach(tabItem => {
  const tabId = tabItem.id as number;
  chrome.tabs.sendMessage(tabId, { action });
})
```

File: src/utils.ts (hunk 2 above)
Comment (AhsanAyaz) [reply src/utils.ts:8]: I'm not clear on what you're referring to. The `AppTab` type? `repo`?  or the `tab` variable etc?

Comment (Skarpej) [reply src/utils.ts:8]: this one ->  let repoName = urlChunks[0];

File: src/pages/Popup/Popup.tsx (hunk 1 above)
Comment (Skarpej) [reply src/pages/Popup/Popup.tsx:38]: Alright, then it looks good!

File: src/utils.ts (hunk 2 above)
Comment (AhsanAyaz) [reply src/utils.ts:8]: Ah, thanks. I fixed it 👍🏽

**Pull request merged**